### ISSUE TITLE
Fargate and Declarative Pipeline issue

### ISSUE THREAD

**Issue by Pietro-G** [jenkinsci/amazon-ecs-plugin#106]
Currently trying to use the plug in to connect to a AWS Fargate server to sping up a Docker Container Slave and Jenkins will not log in to a Cloud no matter what. It will constantly output `ERROR: Cloud does not exist: cloud-default` 

Using Jenkins Ver 2.176.1.
Amazon EC2 Container Service Plug-in Ver 1.20

Log outputs:
[Pipeline] }

[Pipeline] // withEnv

[Pipeline] End of Pipeline

[Bitbucket] Notifying commit build result

[Bitbucket] Build result notified

ERROR: Cloud does not exist: cloud-default

Finished: FAILURE

Failed on stage:
    stage('SSH Slave Test') {
      agent {
              ecs {
                  inheritFrom '<<Tried both Task Definition Over-ride and Normal Label>>'
              }
          }
          steps {
              sh 'echo hello'
          }
    }

**Comment by webratz** [comment]
can you maybe share your configuration for the plugin?

**Comment by Pietro-G** [comment]
[Amazon EC2 Container Config.pdf](https://github.com/jenkinsci/amazon-ecs-plugin/files/3335581/Amazon.EC2.Container.Config.pdf)
`pipeline {
  agent none
  stages {
    stage('SSH Slave Test') {
      agent {
              ecs {
                  cloud 'ecs-slaves'
                  inheritFrom 'fargate-slaves'
              }
          }
          steps {
                sh 'echo hello'
               }
          }
     }
}`

**Comment by vladislavzl** [comment]
@Pietro-G I think you should use inheritFrom 'ecs-slaves' in pipeline.

**Comment by Pietro-G** [comment]
@vladislavzl I gave up on this plug in and use the docker-plugin instead and configured everything manually (the reason being software development cycles have short-term deadlines). It would've been nice to implement Fargate or even EC2 instances - to dynamically provision docker nodes. Sadly I can't provision proprietary company logs, but a lot of the settings that I was trying to pass in the cloud template were being over-written in our AWS structure regardless of how we tweaked security groups and what not.
